cpg_infra/github_wif/driver.py

```diff
@@ -0,0 +1,436 @@
+# flake8: noqa: ANN001,ANN201,ERA001
+"""
+GitHub Workload Identity Federation (WIF) setup for Pulumi
+
+This module provides functionality to set up GitHub repositories with
+GCP Workload Identity Federation for OIDC authentication and Artifact Registry access.
+"""
+
+import re
+from typing import Any, Literal
+
+import pulumi
+import pulumi_gcp as gcp
+import pulumi_github as github
+
+# Constants
+WIF_POOL_NAME = 'github-pool'
+WIF_PROVIDER_NAME = 'github-provider'
+GITHUB_ORG = 'populationgenomics'
+
+# Service account name max length is 30 characters
+SA_NAME_MAX_LENGTH = 30
+
+
+def validate_github_wif_config(config: dict[str, Any]) -> None:
```

Comment:
This validation is fairly minimal, I'd like to move cpg-infra across to using pydantic for validating config in the future, but it would also be ok to use the same way of specifying and parsing config that the main stack uses with the `DeserializableDataclass` https://github.com/populationgenomics/cpg-infrastructure/blob/main/cpg_infra/config/config.py#L42